src/exception-safety.md

```diff
@@ -1,50 +1,101 @@
+<!--
 # Exception Safety
+-->
 
+# 例外安全性
+
+<!--
 Although programs should use unwinding sparingly, there's a lot of code that
 *can* panic. If you unwrap a None, index out of bounds, or divide by 0, your
 program will panic. On debug builds, every arithmetic operation can panic
 if it overflows. Unless you are very careful and tightly control what code runs,
 pretty much everything can unwind, and you need to be ready for it.
+-->
+
+プログラム内では巻き戻しを注意深く使用するべきですが、パニック*し得る*コードが
+たくさんあります。もし None をアンラップしたり、境界外のインデックスを指定したり、
+0 で除算したりしたら、プログラムはパニックするでしょう。デバッグビルドでは、
+全ての算術演算は、オーバーフロー時にパニックします。非常に注意深く、
+そしてどのコードを実行するかを厳しくコントロールしない限り、ほとんどすべての
+コードが巻き戻しをする可能性があり、これに対して準備をする必要があります。
 
+<!--
 Being ready for unwinding is often referred to as *exception safety*
 in the broader programming world. In Rust, there are two levels of exception
 safety that one may concern themselves with:
+-->
+
+巻き戻しに対して準備が出来ていることは、もっと広いプログラミングの世界において、
+しばしば*例外安全*と呼ばれています。 Rust では、プログラムが関わる可能性のある、
+2 つの例外安全のレベルがあります。
 
+<!--
 * In unsafe code, we *must* be exception safe to the point of not violating
   memory safety. We'll call this *minimal* exception safety.
 
 * In safe code, it is *good* to be exception safe to the point of your program
   doing the right thing. We'll call this *maximal* exception safety.
+-->
+
+* アンセーフなコードでは、メモリ安全性を侵害しないという点において、
+  例外安全で*なければなりません*。これを、*最小限*の例外安全と呼びます。
+
+* 安全なコードでは、プログラムが正しいことを行なうという点において、
```

Comment:
same here.

For example, you could always catch any exception and exit the process successfully to make it exception safe. but this is not "dong the right thing"